### PR TITLE
haskell: add all haskell-mode keybindings to literate-haskell-mode

### LINKED ISSUE #3458: literate haskell keybindings
Steps to reproduce using Windows 8.1 and (spacemacs-0.104.2):

1. Create literate haskell file or [download one](http://www.seas.upenn.edu/~cis194/spring13/lectures/01-intro.lhs)
2.  `SPC m` and get `SPC m is undefined`

However things work with `M-x haskell-process-load-or-reload`, which is interesting since earlier I had an issue using interactive Haskell at all in #3443.

### PR DESCRIPTION
**Pull request by jb55**
Rebased and cleaned up version of #3459 by @codygman

Closes #3459
Fixes #3458

Few more things to do

- [x] Test literate haskell + ghc-mod
- [x] Test literate haskell + hindent
- [x] Add `literate-haskell-mode` to `spacemacs/declare-prefix-for-mode`

### CODE REVIEW

**Comment by StreakyCobra** [comment]
Will also fix #3458

**Comment by jb55** [comment]
kk updated description :+1:

**Comment by robbyoconnor** [comment]
This is sexy as hell -- nice work @jb55 :purple_heart: :yellow_heart: :heart: :blue_heart: :green_heart:

**Comment by d12frosted** [comment]
Nice, @jb55 thanks for PR :100:

File: layers/+lang/haskell/packages.el [hunk 1]
```diff
@@ -29,12 +29,15 @@
...
   (defun haskell/init-helm-hoogle ()
     (use-package helm-hoogle
       :defer t
       :init
-      (spacemacs/set-leader-keys-for-major-mode 'haskell-mode "hf" 'helm-hoogle))))
+      (dolist (m haskell-modes)
+        (spacemacs/set-leader-keys-for-major-mode m "hf" 'helm-hoogle)))))
 
 (defun haskell/post-init-flycheck ()
   (spacemacs/add-flycheck-hook 'haskell-mode))
```
Comment (StreakyCobra) [inline layers/+lang/haskell/packages.el]: Just to be sure, flycheck doesn't support literate-haskell?

**Comment by StreakyCobra** [comment]
Lines 53 + 56-66: just to be sure, ghc-mode doesn't support literate-haskell?
Lines 122-128: I think the prefixes' names should also be added in literate-haskell.
Lines 274-275: just to be sure, hindent doesn't support literate-haskell?

**Comment by codygman** [comment]
Thanks @jb55!

@StreakyCobra I don't know offhand, I just know that opening a literate haskell file didn't work at the time I wrote this.

File: layers/+lang/haskell/packages.el (hunk 1 above)
Comment (jb55) [inline layers/+lang/haskell/packages.el]: hmm it should, I'm going to be using this a lot tomorrow so I'll be able to test it a bunch

Comment (jb55) [inline layers/+lang/haskell/packages.el]: I noticed the hlint checker having issues, had to disable it. haskell-ghc seems to work fine. I'll update this line in a bit.

**Comment by jb55** [comment]
@StreakyCobra I don't use hindent or ghc-mod, but now I have an excuse to try them out. I've updated the description to track the progress.

**Comment by syl20bnr** [comment]
:+1: 
Remove [WIP] when you judge it is ready.

**Comment by jb55** [comment]
tested and rebased :+1: 

* couldn't get ghc-mod working, so I'll leave it as is for now 
* hindent doesn't work in literate-haskell-mode
* added prefixes

**Comment by d12frosted** [comment]
> couldn't get ghc-mod working, so I'll leave it as is for now

I thought ghc-mod supports lhs.

**Comment by jb55** [comment]
ok then I'll have to take your word on it since I wasn't able to test it. I've added its bindings to literate-haskell-mode.

That should be it...

**Comment by syl20bnr** [comment]
Thank you :+1: 
Cherry-picked into develop branch, you can safely delete your branch.

**Pull request closed without merging**